Sources/Views/InputTextView.swift

```diff
@@ -118,31 +155,46 @@ open class InputTextView: UITextView {
         layer.cornerRadius = 5.0
         layer.borderWidth = 1.25
         layer.borderColor = UIColor.lightGray.cgColor
-
-        addSubviews()
-        addConstraints()
+        addObservers()
+        addPlaceholderLabel()
     }
-
-    private func addSubviews() {
-
+    
+    /// Adds the placeholderLabel to the view and sets up its initial constraints
+    private func addPlaceholderLabel() {
+        
         addSubview(placeholderLabel)
-    }
-
-    private func addConstraints() {
-
         placeholderLabelConstraintSet = NSLayoutConstraintSet(
             top:    placeholderLabel.topAnchor.constraint(equalTo: topAnchor, constant: placeholderLabelInsets.top),
             bottom: placeholderLabel.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -placeholderLabelInsets.bottom),
             left:   placeholderLabel.leftAnchor.constraint(equalTo: leftAnchor, constant: placeholderLabelInsets.left),
             width:  placeholderLabel.widthAnchor.constraint(equalTo: widthAnchor, constant: -(placeholderLabelInsets.left + placeholderLabelInsets.right))
             ).activate()
     }
-
+    
+    /// Updates the placeholderLabels constraint constants to match the placeholderLabelInsets
     private func updateConstraintsForPlaceholderLabel() {
-
+        
         placeholderLabelConstraintSet?.top?.constant = placeholderLabelInsets.top
         placeholderLabelConstraintSet?.bottom?.constant = -placeholderLabelInsets.bottom
         placeholderLabelConstraintSet?.left?.constant = placeholderLabelInsets.left
         placeholderLabelConstraintSet?.width?.constant = -(placeholderLabelInsets.left + placeholderLabelInsets.right)
     }
+    
+    /// Adds a notification for .UITextViewTextDidChange to detect when the placeholderLabel
+    /// should be hidden or shown
+    private func addObservers() {
+        
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(InputTextView.textViewTextDidChange),
+                                               name: Notification.Name.UITextViewTextDidChange,
+                                               object: nil)
+    }
+    
+    // MARK: - Notifications
+    
+    /// Updates the placeholderLabel's isHidden property based on the text being empty or not
+    @objc
+    open func textViewTextDidChange() {
+        placeholderLabel.isHidden = !text.isEmpty
```

Comment:
Actually we should probably move this back into a `didSet`? Too easy to override the behavior?

Reply:
sure